cerberus-ip-translator-lambda/src/main/java/com.nike.cerberus.lambda.waf/CerberusMetadataLookup.java

```diff
@@ -98,26 +108,16 @@ private OkHttpClient createHttpClient() {
         ArrayList<String> owner = new ArrayList<>();
 
         for (Map<String, String> entry : sdbMetadata) {
+            if (entry.get("name").equals(sdbName) && !sdbName.isEmpty()) {
+                return getOwner(owner, entry);
+            }
+        }
 
-            if (entry.get("path").contains(sdbName) && !sdbName.isEmpty()) {
-                owner.add(entry.get("owner"));
-                if (entry.get("created_by").contains(EMAIL_SYMBOL)) owner.add(entry.get("created_by"));
-                if (entry.get("last_updated_by").contains(EMAIL_SYMBOL)
-                        && !entry.get("last_updated_by").equals(entry.get("created_by"))) {
-                    owner.add(entry.get("last_updated_by"));
-                }
-                return owner;
-            } else {
-                if (entry.containsValue(principalName)) {
-                    owner.add(entry.get("owner"));
-                    if (entry.get("created_by").contains(EMAIL_SYMBOL)) owner.add(entry.get("created_by"));
-                    if (entry.get("last_updated_by").contains(EMAIL_SYMBOL)
-                            && !entry.get("last_updated_by").equals(entry.get("created_by"))) {
-                        owner.add(entry.get("last_updated_by"));
-                    }
-                    return owner;
-                }
+        for (Map<String, String> entry : sdbMetadata) {
```

Comment:
Is there a reason you created another for loop? Can this if statement be added to the for loop on line 110?

Reply:
During testing I ran into a potential case where the principal arn matches the iam principal of a different sdb further up the list from the one I'm actually looking for. The two loops are to double filter, and give precedence to filtering on the name and then filtering on the principal if nothing was found.

Reply:
if the owner is found the function will exit so there shouldn't need to be an additional if statement

Reply:
Yes! Because of the first return statement, okay cool